app/models/suggest_query.rb

```diff
@@ -26,11 +26,51 @@ def find_record_by_query_key!
     result
   end
 
-  def find_cached_block
-    block = Block.cached_find(query_key)
-    raise Api::V1::Exceptions::BlockNotFoundError if block.blank?
+  def aggregate_query!
+    results = Hash.new { |h| h[:data] = Array.new }
+
+    # If query_key is all numbers, search block
+    if QueryKeyUtils.integer_string?(query_key) && (block = find_cached_block).present?
+      results[:data] << block.serializable_hash[:data]
+      return results
+    end
+
+    # If the string length is less than 2, the query result will be empty
+    raise ActiveRecord::RecordNotFound if query_key.length < 2
+
+    if QueryKeyUtils.valid_hex?(query_key)
+      query_methods.each { results[:data] << _1.call.serializable_hash[:data] if _1.call.present? }
+    end
+    if QueryKeyUtils.valid_address?(query_key) && (address = find_cached_address).present?
+      results[:data] << address.serializable_hash[:data]
+    end
+    if (udts = find_udts_by_name_or_symbol).present?
+      results[:data].concat(udts.serializable_hash[:data])
+    end
+    if (collections = find_nft_collections_by_name).present?
+      results[:data].concat(collections.serializable_hash[:data])
+    end
+
+    raise ActiveRecord::RecordNotFound if results.blank?
```

Comment:
Maybe returns null array is more simple.You can discuss with frontend developer.